src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -969,52 +972,35 @@ void readAnalyze(TrackPointer track, double sampleRate, int timingOffset, bool i
         }
     }
 
-    if (memoryCuesLoops.size() > 0) {
-        std::sort(memoryCuesLoops.begin(), memoryCuesLoops.end(), [](const memory_cue_loop_t& a, const memory_cue_loop_t& b) -> bool {
+    if (memoryCuesAndLoops.size() > 0) {
+        std::sort(memoryCuesAndLoops.begin(), memoryCuesAndLoops.end(), [](const memory_cue_loop_t& a, const memory_cue_loop_t& b) -> bool {
             return a.startPosition < b.startPosition;
         });
 
         bool mainCueFound = false;
-        bool firstLoopFound = false;
-
-        // Add memory cues and loops
-        for (int memoryCueLoopIndex = 0; memoryCueLoopIndex < memoryCuesLoops.size(); memoryCueLoopIndex++) {
-            memory_cue_loop_t memoryCueLoop = memoryCuesLoops[memoryCueLoopIndex];
+        
+	// Add memory cues and loops
+        for (int memoryCueOrLoopIndex = 0; memoryCueOrLoopIndex < memoryCuesAndLoops.size(); memoryCueOrLoopIndex++) {
+            memory_cue_loop_t memoryCueOrLoop = memoryCuesAndLoops[memoryCueOrLoopIndex];
 
-            if (memoryCueLoop.endPosition == -1) {
+            if (memoryCueOrLoop.endPosition == Cue::kNoPosition) {
                 if (mainCueFound) {
                     // Add remaining memory cues as hot cues (after actual found hotcues) as Mixxx can only have 1 cue
-                    setHotCue(track, memoryCueLoop.startPosition, -1, ++lastHotCueIndex, memoryCueLoop.comment, mixxx::RgbColor::nullopt(), memoryCueLoop.colorID);
+                    setHotCue(track, memoryCueOrLoop.startPosition, -1, ++lastHotCueIndex, memoryCueOrLoop.comment, mixxx::RgbColor::nullopt(), memoryCueOrLoop.colorID);
```

Comment:
Sorry I did not catch this before, but do not use `++` or `--` to assign and use a value simultaneously. IMO this is an antifeature of C++; it's quite confusing. Do `++lastHotCueIndex` on its own line, then `lastHotCueIndex`.